src/decorator.js

```diff
@@ -0,0 +1,16 @@
+import shouldPureComponentUpdate from './function';
+import { Component } from 'react';
+
+export default function PureRender(Component) {
+  const componentName = Component.name || 'component';
+
+  if (!(Component.prototype instanceof Component)) {
+    throw new Error(`${componentName} is not a React component`);
```

Comment:
>Don't all React components have to inherit from Component?

Nope. You can totally not inherit, as long as you don't need `forceUpdate` or `setState`. I do it all the time.

Also, why not allow it on `createClass`-style components? It should work, no?